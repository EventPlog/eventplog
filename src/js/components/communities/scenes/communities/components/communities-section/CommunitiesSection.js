import React from 'react'
import { Link } from 'react-router-dom'


// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Pagination from 'js/components/shared/pagination'

export const generateTitle = (community = {}) => (
  <Link to={`/communities/${community.id}`}>
    {community.name}
  </Link>
)

export const generateMeta = (community = {}) => (
  <ul>
    <li>
      {community.no_of_members} members
    </li>
    <li>
      {community.no_of_upcoming_events} upcoming event
    </li>
    <li>
      <span className="meta-label">Interests</span>: {community.interests && community.interests.map(interest => <span className="hashtag">{interest}</span> )}
    </li>
  </ul>
)

const CommunitySection = ({
  title,
  showCTA = true,
  communities = {data: [], meta: {}},
  getCommunities,
  followCommunity,
}) => {

  const {loading, error, data = [], meta = {}} = communities;

  if (loading) {
    return <Loading />
  }

  if (error) {
    return <Error msg={communities.error} />
  }

  return (
    <ContentPanel title={title}>
      {data && data.map(({description, featured_image, ...community}) => {
          const title = generateTitle(community)
          const meta = generateMeta(community)
          return (
            <Link to={`/communities/${community.id}`}>
              <ContentPanel.Card
                key={community.id}
                {...{title, description, featured_image, meta}}
                showButton={showCTA && !community.joined}
                btn={{onClick: followCommunity, text: 'Follow'}} />
            </Link>
          )
        }
      )}
      {data && data.length < 1 && <p>This hall for communities seem empty ...</p>}
      {
        meta && meta.total_pages && data.length > 0
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getCommunities} />
          : ''
      }
    </ContentPanel>
  )
}

export default CommunitySection