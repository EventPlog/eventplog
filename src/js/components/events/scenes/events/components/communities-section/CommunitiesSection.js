import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

// internal
import Sidebar from 'js/components/shared/sidebar'

export const generateTitle = (community) => (
  <Link to={`/communities/${community.id}/`}>
    {community.name}
  </Link>
)

export const generateDescription = (focus) => (
  `${focus} community`
)

export const generateMeta = (community) => (
  `${community.no_of_members} members`
)


const CommunitiesSection = ({ communities }) => (
  <Sidebar title="Communities suggestions">
    {communities && communities.map(({featured_image, ...community}) => {
        const title = generateTitle(community);
        const description = generateDescription(community.focus)
        const meta = generateMeta(community)
        const btnText = 'Join'
        return <Sidebar.Card {...{title, description, featured_image, btnText, meta}} />
      }
    )}
  </Sidebar>
)

export default CommunitiesSection