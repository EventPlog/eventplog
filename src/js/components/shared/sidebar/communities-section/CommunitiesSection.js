import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'

export const generateTitle = (community) => (
  <Link to={`/communities/${community.id}/`}>
    {community.name}
  </Link>
)

export const generateDescription = (interest) => (
  `${interest || 'Generic'} community`
)

export const generateMeta = (community) => (
  `${community.no_of_followers} followers`
)

const CommunitiesSection = ({
  title,
  communities = {data: [], meta: {}},
  followCommunity
}) => {
  if (communities.loading) {
    return <Loading />
  }
  if (communities.error) {
    return <Error msg={communities.error} />
  }
  return (
    <Sidebar title={title || "Communities suggestions"}>
      {communities.data && communities.data.map(({featured_image, ...community}) => {
          const title = generateTitle(community);
          const description = generateDescription(community.interest)
          const meta = generateMeta(community)
          const btn = community.following
                        ? {}
                        : {onClick: () => {followCommunity(community)}, text: 'Follow'}
          return (
            <Link to={`/communities/${community.id}/`}>
              <Sidebar.Card key={community.id} {...{title, description, featured_image, btn, meta}} />
            </Link>
          )
        }
      )}
    </Sidebar>
  )
}

export default CommunitiesSection