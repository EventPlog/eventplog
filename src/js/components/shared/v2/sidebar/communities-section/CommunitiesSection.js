import React from 'react'
import { Icon } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import styled, { css } from 'styled-components'

// internal
import Sidebar from 'js/components/shared/v2/sidebar'
import Loading from 'js/components/shared/loading'
import { pluralize, genCommunityLink, genEventLink, titleize } from 'js/utils'

export const generateTitle = (community) => (
  <Link to={genCommunityLink(community)}>
    {community.name}
  </Link>
)

export const generateDescription = (interests = []) => (
  interests && interests.length > 0 ? `Interested in ${interests.join(', ')}` : ''
)

export const generateMeta = ({ no_of_followers }) => (
  [
    <span className="count">{no_of_followers}</span>,
    titleize(pluralize('member', no_of_followers))
  ]
)

const styles = css`
  .see-more {
    position: relative
    font-size: 1rem;
    padding-bottom: 1.2rem;
    text-align: center;
    color: ${props => props.theme.activeLink};

    span {
      padding-left: 8px;
      position: absolute;
      top: 2px;
    }
  }
`
const CommunitiesSection = ({
  title,
  communities = {data: [], meta: {}},
  followCommunity,
  slug
}) => {
  const { loading, error, data = []} = communities
  return (
    <Sidebar title={title || "Communities you may know"}>
      {loading && <Loading />}
      {error && <Loading.Error />}
      {data && data.map(({featured_image, ...community}) => {
        console.log(community)
          const title = generateTitle(community);
          const description = generateDescription(community.topic_interests)
          const meta = generateMeta(community)
          const titleLink = genCommunityLink(community)
          const { no_of_followers, brand_color } = community
          const btn = community.following
                        ? {}
                        : {onClick: () => {followCommunity(community)}, text: 'Follow'}
          return (
            <Sidebar.Card key={community.id} {...{title, description, featured_image, btn, meta, titleLink, no_of_followers, brand_color}} />
          )
        }
      )}

      { data && data.length > 4 &&
          <Link to="/communities">
            <div className="see-more">
              See more communities <span><Icon name="angle right" /></span>
            </div>
          </Link>
        } 
    </Sidebar>
  )
}

export default styled(CommunitiesSection)`${styles}`