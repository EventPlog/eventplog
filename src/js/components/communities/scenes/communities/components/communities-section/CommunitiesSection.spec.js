import React from 'react';
import CommunitiesSection from './CommunitiesSection';
import { shallow } from 'enzyme';

describe('CommunitiesSection', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <CommunitiesSection/> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.communities-section').length).toEqual(1);
  })
});