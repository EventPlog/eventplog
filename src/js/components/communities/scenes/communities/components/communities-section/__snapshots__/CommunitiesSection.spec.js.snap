// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Communities::Communities::CommunitiesSection should render correctly 1`] = `
<ContentPanel
  title="Events Section"
>
  <MainContentCard
    btn={
      Object {
        "onClick": [Function],
        "text": "Follow",
      }
    }
    description="A community of developers united in knowledge sharing."
    featured_image="/login-bg.jpg"
    key="1"
    meta={
      <ul>
        <li>
           
          followers
        </li>
        <li>
          2
           upcoming 
          events
        </li>
        <li>
          <span
            className="meta-label"
          >
            Interests
          </span>
          : 
          <span
            className="hashtag"
          >
            Technology
          </span>
          <span
            className="hashtag"
          >
            Software Engineering
          </span>
        </li>
      </ul>
    }
    showButton={false}
    title={
      <Link
        replace={false}
        to="/communities/1"
      >
        Facebook Developer Circle Lagos
      </Link>
    }
    titleLink="/communities/1"
  />
  <MainContentCard
    btn={
      Object {
        "onClick": [Function],
        "text": "Follow",
      }
    }
    description="Developer community bringing up community interest, powereed by google"
    featured_image="/tech_is_in_you.png"
    key="2"
    meta={
      <ul>
        <li>
           
          followers
        </li>
        <li>
          2
           upcoming 
          events
        </li>
        <li>
          <span
            className="meta-label"
          >
            Interests
          </span>
          : 
          <span
            className="hashtag"
          >
            Developers
          </span>
        </li>
      </ul>
    }
    showButton={true}
    title={
      <Link
        replace={false}
        to="/communities/2"
      >
        Google Developer Group
      </Link>
    }
    titleLink="/communities/2"
  />
  <MainContentCard
    btn={
      Object {
        "onClick": [Function],
        "text": "Follow",
      }
    }
    description="A community of developers united in knowledge sharing"
    featured_image="/login-bg.jpg"
    key="3"
    meta={
      <ul>
        <li>
           
          followers
        </li>
        <li>
          2
           upcoming 
          events
        </li>
        <li>
          <span
            className="meta-label"
          >
            Interests
          </span>
          : 
          <span
            className="hashtag"
          >
            Technology
          </span>
        </li>
      </ul>
    }
    showButton={false}
    title={
      <Link
        replace={false}
        to="/communities/3"
      >
        ForLoop Lagos
      </Link>
    }
    titleLink="/communities/3"
  />
</ContentPanel>
`;
